server/services/omnichannel-voip/service.ts

```diff
@@ -1,10 +1,93 @@
-import { ILivechatVoipService } from '../../sdk/types/IOmnichannelVoipService';
+import { Db } from 'mongodb';
+import _ from 'underscore';
+
+import { IOmnichannelVoipService } from '../../sdk/types/IOmnichannelVoipService';
 import { ServiceClass } from '../../sdk/types/ServiceClass';
+import { IVoipConnectorResult } from '../../../definition/IVoipConnectorResult';
+import { IVoipExtensionBase } from '../../../definition/IVoipExtension';
+import { Logger } from '../../lib/logger/Logger';
+import { Voip } from '../../sdk';
+import { IAgentExtensionMap, IOmnichannelVoipServiceResult } from '../../../definition/ILivechatVoipServiceResult';
+/**
+ * Note (Amol) :
+ * Why cant we get the direct access to the model here and have to work with
+ * raw model? This is still a server code. So ideally it should have access to the model
+ * like the REST apis have.
+ *
+ * Check this during the code review and after code review is done, delete this comment.
+ */
+import { UsersRaw } from '../../../app/models/server/raw/Users';
 
-export class OmnichannelVoipService extends ServiceClass implements ILivechatVoipService {
+export class OmnichannelVoipService extends ServiceClass implements IOmnichannelVoipService {
 	protected name = 'omnichannel-voip';
 
+	private logger: Logger;
+
+	private users: UsersRaw;
+
+	constructor(db: Db) {
+		super();
+		this.users = new UsersRaw(db.collection('users'));
+		this.logger = new Logger('OmnichannelVoipService');
+	}
+
 	getConfiguration(): any {
 		return {};
 	}
+
+	private async getAllocatedExtesionAllocationData(projections: any): Promise<any> {
+		const roles: string[] = ['livechat-agent', 'livechat-manager', 'admin'];
+		const options = {
+			sort: {
+				username: 1,
+			},
+			projection: {
+			},
+		};
+		options.projection = projections;
+		const query = {
+			extension: { $exists: true },
+		};
+		const data = await this.users.findUsersInRolesWithQuery(roles, query, options).toArray();
+		return data;
+	}
+
+	async getAvailableExtensions(): Promise<IOmnichannelVoipServiceResult> {
+		const allExtensions = Promise.await(Voip.getExtensionList()) as IVoipConnectorResult;
```

Comment:
```suggestion
		const allExtensions = await Voip.getExtensionList();
```

Reply:
Since its already a typescript file, we can use this shortcut instead of calling Promise.await.

Also the return type of this method is already sepcified, so we don't have to override the type

Reply:
Fixed